### PR TITLE
Refactor JS code for all JS files for uniformity

### PR DESCRIPTION
**Pull request by jermspeaks**
- Initialized variables on top for performance
- Used function form for every file
- Refactored massive functions into smaller, manageable functions (for readability and clarity)
- Added comments to all functions
- Note: not entirely tested. Couldn't figure out how to do post-email button interactions in DEV environment. Tested everything else out in Chrome. Unsure about Firefox, Safari and IE.
- Note: couldn't figure out how to squash my commits after pushing. If someone wants to help me out with that, it would be much appreciated

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -24,3 +24,4 @@
 .DS_Store
 
 /coverage/
+Gemfile.lock
```
Comment (mehulkar) [inline .gitignore]: Gemfile.lock should not be ignored

**Comment by mehulkar** [comment]
Hey @jermspeaks thanks for the work. Few points of feedback:

- Leave `Gemfile.lock` in. It's good practice to included the locked versions in the project
- Let's leave the `cities.js` file as is right now. I'm planning to revert that back to server side rendered and I don't want to change/refactor things when we don't have tests in place because it would require extensive testing. i.e. I don't want to change something that works. 
- kind of nitpick, but can you change your editor to use 2 spaces instead of tabs? In Sublime, you can click `View > Indentation`

Once you add commits to do these things, with something like this: (there's a few thing that could go wrong here, ping me if you need help)

```bash
git checkout -b js-refactor
git reset --soft upstream/master
git add .
git commit -m "Refactor JS" # or whatever commit message
```

**Comment by jermspeaks** [comment]
Thanks @mehulkar. I'm using Atom, so I'll figure out how to change the indentation to spaces instead of tabs. If I have any questions about your points, I'll ping you.

**Comment by jermspeaks** [comment]
Created new PR, squashing other commits.

**Pull request closed without merging**